### ISSUE TITLE
[Feature] Add limited visibility to repos

### ISSUE THREAD

**Issue by bestlinuxgamers** [go-gitea/gitea#15331]
## Description
I would like to make repos available only to logged in users, as is already possible for organizations.
This is currently possible with the REQUIRE_SIGNIN_VIEW setting, but I don't want this to affect all repos.

## What to implement
A good solution would be to add a "Limited" visibility (like for organizations).

**Comment by xeruf** [comment]
Yes I agree, I don't want to add all our team members who have to log in via SSO anyways to an organization, this would be very handy!
